src/credentials/FabricTable.cpp

```diff
@@ -2180,4 +2181,72 @@ CHIP_ERROR FabricTable::SetShouldAdvertiseIdentity(FabricIndex fabricIndex, Adve
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR FabricTable::FetchVIDVerificationStatement(FabricIndex fabricIndex, MutableByteSpan & outVIDVerificationStatement) const
+{
+    VerifyOrReturnError(mOpCertStore != nullptr, CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrReturnError(outVIDVerificationStatement.size() >= kVendorIdVerificationStatementV1Size, CHIP_ERROR_BUFFER_TOO_SMALL);
+
+    // TODO(#38308): Add VIDVerificationStatement loading support.
+    outVIDVerificationStatement.reduce_size(0);
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR FabricTable::FetchVVSC(FabricIndex fabricIndex, MutableByteSpan & outVVSC) const
+{
+    VerifyOrReturnError(mOpCertStore != nullptr, CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrReturnError(outVVSC.size() >= kMaxCHIPCertLength, CHIP_ERROR_BUFFER_TOO_SMALL);
+
+    // TODO(#38308): Add VVSC loading support.
+    outVVSC.reduce_size(0);
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR FabricTable::SignVIDVerificationRequest(FabricIndex fabricIndex, const ByteSpan & clientChallenge,
```

Comment:
should the spans be passed by value instead of const reference? They are basically "pointer + size" ... unsure if there is a benefit on space but it would make code shorter.

Our SDK is all over the place .. CaseSession.h and NetworkCommissioningDriver.h use by-value, where as others use const reference.

Pigweed has it's own version and they recommend by value to avoid indirection: https://pigweed.dev/pw_span/#guides

Reply:
Will do in follow-up.